packages/image-block/src/components/ImageCaption.tsx

```diff
@@ -20,40 +21,42 @@ export const ImageCaption = ({
 }: ImageCaptionProps) => {
     const { designTokens } = useGuidelineDesignTokens();
 
+    const rawName = JSON.stringify(parseRawValue({ raw: name ?? '' }));
+    const nameHtml = serializeRawToHtml(rawName, designTokens);
+    const rawDescription = JSON.stringify(parseRawValue({ raw: description ?? '' }));
+    const descriptionHtml = serializeRawToHtml(rawDescription, designTokens);
+
     return (
-        <div className="tw-flex tw-flex-col tw-justify-center tw-items-center tw-text-center tw-mt-3 tw-gap-1">
-            <div
-                className={joinClassNames([
-                    'tw-text-text tw-text-[12px] tw-leading-[18px] tw-font-medium',
-                    hasRichTextValue(name) || isEditing ? 'tw-block' : 'tw-hidden',
-                ])}
-            >
-                <RichTextEditor
-                    value={name}
-                    border={false}
-                    readonly={!isEditing}
-                    onBlur={onNameChange}
-                    onTextChange={onNameChange}
-                    designTokens={designTokens}
-                    placeholder={isEditing ? 'Asset name' : undefined}
-                />
-            </div>
-            <div
-                className={joinClassNames([
-                    'tw-text-text-weak tw-text-[11px] tw-leading-[13px]',
-                    hasRichTextValue(description) || isEditing ? 'tw-block' : 'tw-hidden',
-                ])}
-            >
-                <RichTextEditor
-                    value={description}
-                    border={false}
-                    readonly={!isEditing}
-                    onTextChange={onDescriptionChange}
-                    onBlur={onDescriptionChange}
-                    designTokens={designTokens}
-                    placeholder={isEditing ? 'Add a description here' : undefined}
-                />
-            </div>
+        <div className="tw-mt-3 tw-gap-1 tw-flex-1 tw-w-full">
+            {isEditing ? (
+                <>
+                    <RichTextEditor
+                        id="image-title"
```

Comment:
`id` should be unique even if there are multiple blocks on the same page. You can use `appBridge.getBlockId().toString()` and extend the string with `_imageTitle` or something like this.

Reply:
changed 👍